examples/azure-service-mesh/security/cert-manager-letsencrypt/readme.md

```diff
@@ -0,0 +1,276 @@
+# Cert-Manager Let's Encrypt Integration with Istio-based service mesh add-on
+
+This document contains instructions on how to integrate Istio-based service mesh add-on for AKS with cert-manager and obtain let's encrypt certificates for setting up secure ingress gateways. 
+
+## Objectives
+* Deploy bookinfo demo app, expose a secure HTTPS service using simple TLS.
+* Demonstrate HTTPS connections for Azure Service Mesh workloads using cert-manager and let's encrypt as the certificate authority.
+
+## Before you begin
+* [Install](https://learn.microsoft.com/en-us/azure/aks/istio-deploy-addon#install-istio-add-on) Istio-based service mesh add-on on your cluster.
+```shell
+az aks mesh enable -g <rg-name> -n <cluster-name>
+```
+* [Enable external ingressgateway](https://learn.microsoft.com/en-us/azure/aks/istio-deploy-ingress#enable-external-ingress-gateway)
+```shell
+az aks mesh enable-ingress-gateway -g <rg-name> -n <cluster-name> --ingress-gateway-type external
```

Comment:
Would internal ingress gateway also work or does it have to be an external IP? If so I think we should add a note here and a comment in configmap.yaml that if they're using internal, then they'd need to update the ingressService and ingressSelector accordingly.

Reply:
`ingressSelector` and `ingressService` needs to be set to external for ACME challenge path resolved.

Reply:
Internal would not work because http challenge won't be served properly.